docs/runtime-contract.md

```diff
@@ -126,9 +126,9 @@ the OCI specification as long as:
   contents from a particular execution. Because containers (particularly failing
   containers) can experience frequent starts, operators or platform providers
   SHOULD limit the total space consumed by these failures.
-- A container should write its own termination message to `/dev/termination-log`
+- A container SHOULD write its own termination message to `/dev/termination-log`
   by default. If no message is written by the container, the last few lines of
-  log output should be reported as the execution error (i.e. by
+  log output SHOULD be reported as the execution error (i.e. by
```

Comment:
We don't do this today and I don't think we plan to do this. K8s exposes this on the Pod which is not part of the Knative API and is difficult to aggregate up to the Revision.

Reply:
hmm, who is supposed to "report this error" ? Knative? the OCI runtime?

Reply:
I have read it in the past as Knative. A Knative user does not have interaction with the OCI runtime. A user is able to see what is visible through our API, monitoring/logging extension points, anything observable within the container, and anything observable about the lifecycle of the container.

If this is only visible in the OCI API then it is not visible to the end user and is thus could be omitted from the runtime without consequence to user experience.

Reply:
See the discussion in #4035 -- it turns out that this information is collected at the Pod level, but is not aggregated beyond that level.

Having looked into it further, I'm concerned about the cascading update scenario that aggregating this to status information in the apiserver above the Pod level enables. Consider 1000 containers which crash in a coordinated manner -- that's a lot of extra updates (and possibly conflicts which need to be retried) that put load on the apiserver at exactly the wrong time.

Reply:
I think the discussion in #4035 is leaning towards removing this text altogether -- there's no place in the API for these errors to be surfaced beyond the Pod level, and Pods are really ephemeral, so the termination message isn't so helpful.

Reply:
I'm trying to limit the semantic changes made to this PR but I'll remove this text anyway since you guys are convinced it's wrong. I removed the entire bullet since if I remove the last sentence then there's no normative text left - so there's no point in keeping the rest of it.